hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SelectiveWorkBalancerTest.java

```diff
@@ -0,0 +1,194 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.selective.WorkBalancingResult;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.selective.SelectiveWorkBalancer;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.IntStream;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SelectiveWorkBalancerTest {
+
+    private static int CONSUMERS_PER_SUBSCRIPTION = 2;
+    private static int MAX_SUBSCRIPTIONS_PER_CONSUMER = 2;
+
+    private SelectiveWorkBalancer workBalancer = new SelectiveWorkBalancer(CONSUMERS_PER_SUBSCRIPTION, MAX_SUBSCRIPTIONS_PER_CONSUMER);
+
+    @Test
+    public void shouldPerformSubscriptionsCleanup() {
+        // given
+        List<SubscriptionName> subscriptions = someSubscriptions(1);
```

Comment:
i think we could move all those private methods into some fixture builder, i.e: `fixture.withAnonymousSupervisors(2).withSomeSubscriptions(2).build()` etc